tests/unit/client.py

```diff
@@ -262,28 +268,69 @@ def avoid_network_race_condition():
     for timeout in range(10):
         que = db.session.query(NetworkingAction).count()
         if (que > 0):
-            time.sleep(1)
+            time.sleep(.5)
         else:
             return
     raise TimeoutError(" Timed out to avoid race condition. ")
 
 
+def port_has_server(url):
+    """ Returns whether a local TCP port is a connectable http server. """
+
+    try:
+        sess = requests.Session()
+        sess.auth = (username, password)
+        sess.get(url + '/projects')
+        return True
+    except:
+        return False
+
+
+def wait_for_service(url, timeout=15):
+    """
+    Waits for a port to become a working http server
+    timeout -- number of seconds to wait (default 60)
+    """
+
+    begin = time.time()
+    while True:
+        try:
+            sess = requests.Session()
+            sess.auth = (username, password)
+            sess.get(url + '/projects')
+            break
+        except requests.exceptions.ConnectionError:
+            if (time.time() - begin) < timeout:
+                time.sleep(.5)
+            else:
+                raise TimeoutError("Client library test server didn't "
+                                   "start in {} seconds".format(timeout))
+
+
 @pytest.fixture(scope="module")
 def create_setup(request):
+    serv_port = 8000
+    url = 'http://127.0.0.1:{}'.format(serv_port)
+
+    assert not port_has_server(url)
+
     dir_names = make_config()
     initialize_db()
-    proc1 = Popen(['haas', 'serve', '8000'])
-    proc2 = Popen(['haas', 'serve_networks'])
-    time.sleep(1)
+    proc1 = Popen(['haas', 'serve', str(serv_port)], stdout=sys.stdout,
+                  stderr=sys.stderr)
+    proc2 = Popen(['haas', 'serve_networks'], stdout=sys.stdout,
+                  stderr=sys.stderr)
+    wait_for_service(url)  # Loop until the server is up. See #770
     populate_server()
 
-    def fin():
-        proc1.terminate()
-        proc2.terminate()
-        proc1.wait()
-        proc2.wait()
-        cleanup(dir_names)
-    request.addfinalizer(fin)
+    yield
```

Comment:
Pytest [recommends](https://docs.pytest.org/en/latest/fixture.html) using this syntax over addfinalizer(). I think it's much cleaner myself.